src/RouteJs/RouteJs.cs

```diff
@@ -67,13 +89,26 @@ private bool AllowRoute(RouteBase route)
 		/// <param name="route">The route</param>
 		/// <returns>Route information</returns>
 		private RouteInfo GetRoute(Route route)
-		{			
+		{
 			var routeInfo = new RouteInfo
 			{
-				Url = route.Url,
+				Url = GetUrl(route.Url),
 				Constraints = new RouteValueDictionary(),
 			};
 
+			if (IsLowerCaseUrls)
+			{
+				// Convert "controller" and "action" defaults to lowercase.
+				var convertedDefaults = route.Defaults
+					.Where(d => s_defaultKeysToConvert.Any(st => string.Equals(d.Key, st, StringComparison.OrdinalIgnoreCase)))
+					.Select(d => new KeyValuePair<string, object>(d.Key, ((string)d.Value).ToLower()))
+					.ToArray();
+				foreach (var item in convertedDefaults)
+				{
+					route.Defaults[item.Key] = item.Value;
```

Comment:
This could probably go in an implementation of `IDefaultsProcessor` rather than here (eg. `LowercaseUrlDefaultsProcessor`). IDefaultsProcessor is already designed to handle extending the processing of route defaults. You just need to register it in the IoC container (`ComponentRegistration.cs`)